src/main/java/com/minecolonies/coremod/entity/ai/minimal/EntityAISickTask.java

```diff
@@ -0,0 +1,498 @@
+import net.minecraft.potion.EffectInstance;
+import net.minecraft.potion.Effects;
+import net.minecraft.state.properties.BedPart;
+import net.minecraft.tags.BlockTags;
+import net.minecraft.util.Hand;
+import net.minecraft.util.SoundEvents;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.TranslationTextComponent;
+import net.minecraft.world.World;
+
+import java.util.EnumSet;
+import java.util.List;
+
+import static com.minecolonies.api.util.constant.Constants.TICKS_SECOND;
+import static com.minecolonies.api.util.constant.GuardConstants.BASIC_VOLUME;
+import static com.minecolonies.api.util.constant.TranslationConstants.NO_HOSPITAL;
+import static com.minecolonies.coremod.entity.ai.minimal.EntityAISickTask.DiseaseState.*;
+import static com.minecolonies.coremod.entity.citizen.citizenhandlers.CitizenDiseaseHandler.SEEK_DOCTOR_HEALTH;
+
+/**
+ * The AI task for citizens to execute when they are supposed to eat.
+ */
+public class EntityAISickTask extends Goal
+{
+    /**
+     * Min distance to hut before pathing to hospital.
+     */
+    private static final int MIN_DIST_TO_HUT = 5;
+
+    /**
+     * Min distance to hospital before trying to find a bed.
+     */
+    private static final int MIN_DIST_TO_HOSPITAL = 3;
+
+    /**
+     * Min distance to the hospital in general.
+     */
+    private static final long MINIMUM_DISTANCE_TO_HOSPITAL = 10;
+
+    /**
+     * Required time to cure.
+     */
+    private static final int REQUIRED_TIME_TO_CURE = 60;
+
+    /**
+     * Chance for a random cure to happen.
+     */
+    private static final int CHANCE_FOR_RANDOM_CURE = 10;
+
+    /**
+     * Attempts to position right in the bed.
+     */
+    private static final int GOING_TO_BED_ATTEMPTS = 20;
+
+    /**
+     * The waiting ticks.
+     */
+    private int waitingTicks = 0;
+
+    /**
+     * The bed the citizen is sleeping in.
+     */
+    private BlockPos usedBed;
+
+    /**
+     * The different types of AIStates related to eating.
+     */
+    public enum DiseaseState
+    {
+        IDLE,
+        CHECK_FOR_CURE,
+        GO_TO_HUT,
+        SEARCH_HOSPITAL,
+        GO_TO_HOSPITAL,
+        WAIT_FOR_CURE,
+        FIND_EMPTY_BED,
+        APPLY_CURE
+    }
+
+    /**
+     * The citizen assigned to this task.
+     */
+    private final EntityCitizen citizen;
+
+    /**
+     * The state the task is in currently.
+     */
+    private DiseaseState currentState = IDLE;
+
+    /**
+     * Restaurant to which the citizen should path.
+     */
+    private BlockPos placeToPath;
+
+    /**
+     * Delay ticks.
+     */
+    private int delayTicks = 0;
+
+    /**
+     * Instantiates this task.
+     *
+     * @param citizen the citizen.
+     */
+    public EntityAISickTask(final EntityCitizen citizen)
+    {
+        super();
+        this.citizen = citizen;
+        this.setMutexFlags(EnumSet.of(Flag.MOVE));
+    }
+
+    @Override
+    public boolean shouldExecute()
+    {
```

Comment:
I think this AI could be quite a bit easier with all its states/timers by using a statemachine, like the Child minimal AI does

Reply:
I don't think we should add a new state machine to each individual minimal task. Else we'd setup a state machine all these can use somehow (avoid duplicate code)

Reply:
ye using a statemachine is a little bit overkill(though they're cheap) but lateron when we combine the AI's nicely we got AITargets to use already

Reply:
ye